engine/function/src/templates/Basic.ftl

```diff
@@ -1585,6 +1642,52 @@ public class Basic {
         return ifelse(new BooleanVectorDirect(condition), new ${pt.dbArrayDirect}(trueCase), new ${pt.dbArrayDirect}(falseCase));
     }
 
+    /**
+     * Returns elements from either trueCase or falseCase, depending on condition.
+     *
+     * @param condition a boolean value used to select output values.
+     * @param trueCase value returned when condition is true.
+     * @param falseCase value returned when condition is false.
+     * @return An array of ${pt.primitive} whose values are determined by the corresponding elements of condition, trueCase, and falseCase.
+     *         The result element will be trueCase if the condition element is true;
+     *         falseCase if the condition element is false; or the Deephaven null constant if the condition element is null.
+     *         Returns null if any of the inputs is null.
+     */
+    public static ${pt.primitive}[] ifelse(BooleanVector condition, ${pt.primitive} trueCase, ${pt.primitive} falseCase) {
+        if (condition == null) {
+            return null;
+        }
+
+        final int n_c = condition.intSize("condition");
+
+        final ${pt.primitive}[] result = new ${pt.primitive}[n_c];
+
+        for (int i=0; i < n_c; i++) {
+            result[i] = condition.get(i) == null ? ${pt.null} : (condition.get(i) ? trueCase : falseCase);
```

Comment:
same as above regarding prefer not to repeat the evaluation of `condition.get(i)`